src/createReconciler.lua

```diff
@@ -66,13 +72,29 @@ local function createReconciler(renderer)
 			internalAssert(Type.of(virtualNode) == Type.VirtualNode, "Expected arg #1 to be of type VirtualNode")
 		end
 
+		virtualNode.updateChildrenCount = virtualNode.updateChildrenCount + 1
+
+		local currentUpdateChildrenCount = virtualNode.updateChildrenCount
+
 		local removeKeys = {}
 
 		-- Changed or removed children
 		for childKey, childNode in pairs(virtualNode.children) do
 			local newElement = ElementUtils.getElementByKey(newChildElements, childKey)
 			local newNode = updateVirtualNode(childNode, newElement)
 
+			-- If updating this node has caused a component higher up the tree to re-render
+			-- and updateChildren to be re-entered for this virtualNode then
+			-- this result is invalid and needs to be disgarded.
+			if config.tempFixUpdateChildrenReEntrancy then
+				if virtualNode.updateChildrenCount ~= currentUpdateChildrenCount then
+					if newNode and not newNode ~= virtualNode.children[childKey] then
```

Comment:
Looks like some double negatives here? Is this the same meaning, and is it clearer?
```suggestion
					if newNode ~= nil and newNode == virtualNode.children[childKey] then
```

Reply:
Ah right, this actually should be just newNode ~= virtualNode.children[childKey]. I originally had it as 
not newNode == virtualNode.children[childKey] and must have messed up when changing it. I'll see can I add a test that would catch this issue.

Reply:
Added a new test that catches this bug